haystack/document_stores/search_engine.py

```diff
@@ -21,6 +22,10 @@
 from haystack.nodes.retriever import DenseRetriever
 from haystack.utils.scipy_utils import expit
 
+if typing.TYPE_CHECKING:
+    from opensearchpy import OpenSearch
+    from elasticsearch import Elasticsearch
```

Comment:
Why import them only when type checking?

Reply:
Because we just need them as type in `_do_bulk`, we don't actually use them. I think we can also remove the `if typing.TYPE_CHECKING:`.

Reply:
You're using strings in the Union, I guess we can just not import them?